### PR TITLE
Salvage magnet shows cooldown time on Examine

### PR DESCRIPTION
**Pull request by dmnct**
## About the PR 

**Media**

image
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- tweak: Salvage magnet shows remaining cooldown time when examined.

### CODE REVIEW

File: Content.Server/Salvage/SalvageSystem.cs [hunk 1]
```diff
@@ -141,8 +141,22 @@ private void OnMagnetRemoval(EntityUid uid, SalvageMagnetComponent component, Co
...
         private void OnExamined(EntityUid uid, SalvageMagnetComponent component, ExaminedEvent args)
         {
+            bool _gotGrid = false;
+            var _remainingTime = TimeSpan.Zero;
+
             if (!args.IsInDetailsRange)
                 return;
+
+            if (EntityManager.GetComponent<TransformComponent>(component.Owner).GridUid is EntityUid gridId &&
+                        _salvageGridStates.TryGetValue(gridId, out var salvageGridState))
```
[the remark this answers is not in the document]
Comment (dmnct) [reply Content.Server/Salvage/SalvageSystem.cs]: Good catch, thanks

File: Content.Server/Salvage/SalvageSystem.cs [hunk 2]
```diff
@@ -141,8 +141,22 @@ private void OnMagnetRemoval(EntityUid uid, SalvageMagnetComponent component, Co
 
         private void OnExamined(EntityUid uid, SalvageMagnetComponent component, ExaminedEvent args)
         {
+            bool _gotGrid = false;
+            var _remainingTime = TimeSpan.Zero;
```
Comment (metalgearsloth) [inline Content.Server/Salvage/SalvageSystem.cs:145]: Please use the C# style guide for variable naming (no underscore prefix for locals and use var).

File: Content.Server/Salvage/SalvageSystem.cs [hunk 3]
```diff
@@ -155,19 +169,12 @@ private void OnExamined(EntityUid uid, SalvageMagnetComponent component, Examine
                     args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-releasing"));
                     break;
                 case MagnetStateType.CoolingDown:
-                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-cooling-down"));
+                    if (_gotGrid == true)
```
Comment (metalgearsloth) [inline Content.Server/Salvage/SalvageSystem.cs:172]: Same here on style guide, don't use == true.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Salvage/SalvageSystem.cs (hunk 2 above)
Comment (dmnct) [reply Content.Server/Salvage/SalvageSystem.cs]: Fixed

File: Content.Server/Salvage/SalvageSystem.cs (hunk 3 above)
Comment (dmnct) [reply Content.Server/Salvage/SalvageSystem.cs]: Fixed, thank you

File: Content.Server/Salvage/SalvageSystem.cs [hunk 4]
```diff
@@ -141,8 +141,22 @@ private void OnMagnetRemoval(EntityUid uid, SalvageMagnetComponent component, Co
 
         private void OnExamined(EntityUid uid, SalvageMagnetComponent component, ExaminedEvent args)
         {
+            var gotGrid = false;
+            var remainingTime = TimeSpan.Zero;
+
             if (!args.IsInDetailsRange)
                 return;
+
+            if (EntityManager.GetComponent<TransformComponent>(component.Owner).GridUid is EntityUid gridId &&
```
Comment (ZoldorfTheWizard) [inline Content.Server/Salvage/SalvageSystem.cs:150]: *.Owner is deprecated, use the entity UID here

**Review by ZoldorfTheWizard — CHANGES_REQUESTED** [review]

File: Content.Server/Salvage/SalvageSystem.cs (hunk 4 above)
Comment (mirrorcult) [inline Content.Server/Salvage/SalvageSystem.cs:150]:
```suggestion
            if (Transform(uid).GridUid is EntityUid gridId &&
```

Comment (mirrorcult) [reply Content.Server/Salvage/SalvageSystem.cs:150]: as zoldorf mentioned

**Comment by deltanedas** [comment]
revived it :rocket:

**Comment by AJCM-git** [comment]
dupe

**Pull request closed without merging**